rclpy/rclpy/publisher.py

```diff
@@ -100,3 +101,21 @@ def assert_liveliness(self) -> None:
         """
         with self.handle:
             _rclpy.rclpy_assert_liveliness(self.handle)
+
+    def wait_for_all_acked(self, timeout: Duration = Duration(seconds=-1)) -> bool:
+        """
+        Wait until all published message data is acknowledged or until the timeout elapses.
+
+        If the timeout is negative then this function will block indefinitely until all published
+        message data were acknowledged.
+        If the timeout is 0 then this function will be non-blocking; checking all published message
+        data were acknowledged, but not waiting.
+        If the timeout is greater than 0 then this function will return after that period of time
+        has elapsed or all published message data were acknowledged.
+
+        :param timeout: the duration to wait for all published message data were acknowledged
+        :returns: true if all published message data were acknowledged before timeout, otherwise
+            false.
+        """
+        with self.handle:
+            return self.__publisher.wait_for_all_acked(timeout._duration_handle)
```

Comment:
it should use `try&except` statement? Publisher::wait_for_all_acked would throw RCLError exception.

Reply:
After check other codes, I find it also doesn't deal with RCLError.   
e.g.

https://github.com/ros2/rclpy/blob/714f69ee64ee8ea5dbc5f3067211b7886a4e7226/rclpy/rclpy/publisher.py#L59-L73
Not catch exception

The underlying code may throw exception

https://github.com/ros2/rclpy/blob/714f69ee64ee8ea5dbc5f3067211b7886a4e7226/rclpy/src/rclpy/publisher.cpp#L121-L133